scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -2997,6 +2997,16 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
       )
     val tparams = typeParamClauseOpt(ownerIsType = false, ctxBoundsAllowed = true)
     val paramss = paramClauses(ownerIsType = false).require[List[List[Term.Param]]]
+
+    def onlyLastParameterCanBeRepeated(params: List[Term.Param]): Unit = {
+      params.iterator
+        .take(params.length - 1)
+        .filter(p => !p.is[Term.Param.Quasi] && p.decltpe.exists(_.is[Type.Repeated]))
+        .foreach(syntaxError("*-parameter must come last", _))
```

Comment:
```suggestion
        .foreach(syntaxError("vararg parameter must come last", _))
```

Reply:
And I think it should be really `varargs` instead of `vararg` everywhere - sorry!

Reply:
But this is an error taken from scala compiler, do you want to have different error message from what scala reports?

Reply:
We should use the same error message as the compiler, even if it's cryptic. Unrelated, I think the following reads better with a named parameter

```scala
.foreach(p => syntaxError("varargs parameter must come last", p))
```

Term wildcards are great in cases like `foreach(_.foo)` but otherwise it's almost always clearer with a named parameter

Reply:
to be fair @tgodzik must have checked dotty error, because they have different messages:
```
scala> def x(x: Int*, y: Int*): Unit = ???
             ^
       error: *-parameter must come last
```
```
scala> def x(x: Int*, y: Int*): Unit = ???
1 |def x(x: Int*, y: Int*): Unit = ???
  |         ^^^^
  |         varargs parameter must come last
```

Reply:
Updated to not have anonymous lambda in foreach.

I cannot really do `if dotty: errorA else errorB` because we operate on dialects not scala versions and this error not related to any features in dialect. I could use one that I know is only enabled in dotty but that would go against idea of dialects. I would keep current error and maybe change later when dotty gets more mature.